File: app/src/main/java/com/ragdroid/dahaka/activity/home/stats/StatsModel.java
package com.ragdroid.dahaka.activity.home.stats;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import com.ragdroid.dahaka.BR;
import com.ragdroid.dahaka.api.entity.Pokemon;

import java.util.List;

/**
 * Created by garimajain on 16/08/17.
 */

public class StatsModel extends BaseObservable {

    @Bindable
    public String getStats() {
        return stats;
    }

    public void setStats(String stats) {
        this.stats = stats;
        notifyPropertyChanged(BR.stats);
    }

    public String stats;

    public StatsModel(List<Pokemon.Stat> stats) {
        decorateMoves(stats);
    }

    private void decorateMoves(List<Pokemon.Stat> stats) {
        StringBuilder stringBuilder = new StringBuilder();
        stringBuilder.append("Stats : \n");
        for (Pokemon.Stat stat : stats) {
            stringBuilder.append(stat.getStat().getName() + " : " + stat.getBaseStat());
            stringBuilder.append("\n");
        }
        this.stats = stringBuilder.toString();
    }

}

File: app/src/main/java/com/ragdroid/dahaka/activity/home/stats/StatsModel.kt
package com.ragdroid.dahaka.activity.home.stats

import android.databinding.BaseObservable
import android.databinding.Bindable

import com.ragdroid.dahaka.BR
import com.ragdroid.dahaka.api.entity.Pokemon

/**
 * Created by garimajain on 16/08/17.
 */

class StatsModel(stats: List<Pokemon.Stat>) : BaseObservable() {

    @get:Bindable
    var stats: String? = null
    set(stats) {
        field = stats
        notifyPropertyChanged(BR.stats)
    }

    init {
        decorateMoves(stats)
    }

    private fun decorateMoves(stats: List<Pokemon.Stat>) {
        val stringBuilder = StringBuilder()
        stringBuilder.append("Stats : \n")
        for (stat in stats) {
            stringBuilder.append(stat.stat?.name + " : " + stat.baseStat)
            stringBuilder.append("\n")
        }
        this.stats = stringBuilder.toString()
    }

}
